### PR TITLE
Update to require PHP 8.0

### PR DESCRIPTION
**Pull request by ben-challis**
PHP 7.4 is out of support

### CODE REVIEW

File: rector.php [hunk 1]
```diff
@@ -2,26 +2,21 @@
...
     $rectorConfig->parallel();
     $rectorConfig->paths([__DIR__.'/src', __DIR__.'/tests']);
-    $rectorConfig->phpVersion(PhpVersion::PHP_74);
+    $rectorConfig->phpVersion(PhpVersion::PHP_80);
     $rectorConfig->phpstanConfig(__DIR__.'/phpstan-rector.neon');
-    $rectorConfig->skip([
-        DateTimeToDateTimeInterfaceRector::class,
-    ]);
     $rectorConfig->importNames();
-    $rectorConfig->parameters()->set(Option::IMPORT_SHORT_CLASSES, false);
```
Comment (ben-challis) [inline rector.php:21]: Deprecated

File: rector.php [hunk 2]
```diff
@@ -2,26 +2,21 @@
...
 
 return static function (RectorConfig $rectorConfig): void {
     $rectorConfig->parallel();
     $rectorConfig->paths([__DIR__.'/src', __DIR__.'/tests']);
-    $rectorConfig->phpVersion(PhpVersion::PHP_74);
+    $rectorConfig->phpVersion(PhpVersion::PHP_80);
     $rectorConfig->phpstanConfig(__DIR__.'/phpstan-rector.neon');
-    $rectorConfig->skip([
-        DateTimeToDateTimeInterfaceRector::class,
-    ]);
```
Comment (ben-challis) [inline rector.php:19]: Class no longer exists

Comment (marmichalski) [reply rector.php:19]: Might want to add that file to phpstan analysis :see_no_evil:

**Review by marmichalski — DISMISSED** [review]

**Review by marmichalski — APPROVED** [review]

File: rector.php (hunk 2 above)
Comment (ben-challis) [reply rector.php:19]: Yeah good idea, done in https://github.com/Lendable/json-serializer/pull/270/commits/7c5c4f3379b78828015ff0c58632e5bcce9bf0fc

**Pull request merged**